code/_onclick/hud/screen_objects.dm

```diff
@@ -181,7 +181,7 @@
 	cut_overlay(object_overlays)
 	object_overlays.Cut()
 
-/atom/movable/screen/inventory/update_icon()
+/atom/movable/screen/inventory/update_appearance(updates = ALL)
```

Comment:
We should never be overwriting update_appearance at all, this should still be update_icon